src/common/name-syntax.k

```diff
@@ -2,15 +2,16 @@ module NAME-SYNTAX
 imports LIST-SYNTAX
 //todo: why does this pure syntax definition affect krun?
 //PackageName, AmbName and PackageorTypeName syntax doesn't cause problems
-//                     syntax PackageName ::= IdList [klabel('PackageName)]
-//                     syntax ExprName::= Id  [klabel('ExprName)]
+                     syntax PackageName ::= IdList [klabel('PackageName)]
+                     syntax AmbName ::= Id [klabel('AmbName)]
+                                      | AmbName "." Id [klabel('AmbName)]
+                     syntax PackageOrTypeName ::= Id [klabel('PackageOrTypeName)]
+                                                | PackageOrTypeName "." Id [klabel('PackageOrTypeName)]//strict(1)
+syntax KLabel ::= "'MethodName"
```

Comment:
When you discuss it with Radu, if there are such things that need to be added to due to parsing limitations, *please* add a comment next to them saying so and a note that this should be fixed as soon as we have the new parser.  Without that you will not know how or what to improve in the semantics later on.